pallets/nodle-staking/src/lib.rs

```diff
@@ -0,0 +1,1566 @@
+        ) -> Option<Vec<(T::AccountId, ValidatorSnapshot<T::AccountId, BalanceOf<T>>)>> {
+            <Self as pallet_session::SessionManager<_>>::new_session(new_index).map(|validators| {
+                validators
+                    .into_iter()
+                    .map(|v| {
+                        let validator_inst = Self::at_stake(new_index, &v);
+                        (v, validator_inst)
+                    })
+                    .collect()
+            })
+        }
+        fn start_session(start_index: SessionIndex) {
+            <Self as pallet_session::SessionManager<_>>::start_session(start_index)
+        }
+        fn end_session(end_index: SessionIndex) {
+            <Self as pallet_session::SessionManager<_>>::end_session(end_index)
+        }
+    }
+
+    /// Means for interacting with a specialized version of the `session` trait.
+    ///
+    /// This is needed because `Staking` sets the `ValidatorIdOf` of the `pallet_session::Config`
+    pub trait SessionInterface<AccountId>: frame_system::Config {
+        /// Disable a given validator by stash ID.
+        ///
+        /// Returns `true` if new era should be forced at the end of this session.
+        /// This allows preventing a situation where there is too many validators
+        /// disabled and block production stalls.
+        fn disable_validator(validator: &AccountId) -> Result<bool, ()>;
+        /// Get the validators from session.
+        fn validators() -> Vec<AccountId>;
+        /// Prune historical session tries up to but not including the given index.
+        fn prune_historical_up_to(up_to: SessionIndex);
+    }
+
+    impl<T: Config> SessionInterface<<T as frame_system::Config>::AccountId> for T
+    where
+        T: pallet_session::Config<ValidatorId = <T as frame_system::Config>::AccountId>,
+        T: pallet_session::historical::Config<
+            FullIdentification = ValidatorSnapshot<
+                <T as frame_system::Config>::AccountId,
+                BalanceOf<T>,
+            >,
+            FullIdentificationOf = ValidatorSnapshotOf<T>,
+        >,
+        T::SessionHandler: pallet_session::SessionHandler<<T as frame_system::Config>::AccountId>,
+        T::SessionManager: pallet_session::SessionManager<<T as frame_system::Config>::AccountId>,
+        T::ValidatorIdOf: Convert<
+            <T as frame_system::Config>::AccountId,
+            Option<<T as frame_system::Config>::AccountId>,
+        >,
+    {
+        fn disable_validator(
+            validator: &<T as frame_system::Config>::AccountId,
+        ) -> Result<bool, ()> {
+            <pallet_session::Module<T>>::disable(validator)
+        }
+
+        fn validators() -> Vec<<T as frame_system::Config>::AccountId> {
+            <pallet_session::Module<T>>::validators()
+        }
+
+        fn prune_historical_up_to(up_to: SessionIndex) {
+            <pallet_session::historical::Module<T>>::prune_up_to(up_to);
+        }
+    }
```

Comment:
Do we need this?

Reply:
Yes we are using this interface to interact with pallet-session ...

* <pallet_session::Module<T>>::disable(validator) under slashing context.
*  <pallet_session::historical::Module<T>>::prune_up_to(up_to) | Session History management.